doc/index.rst

```diff
@@ -7,14 +7,62 @@ Fast, Scalable and Easy Machine Learning With DAAL4PY
 Daal4py makes your Machine Learning algorithms in Python lightning fast and easy
 to use. It provides highly configurable Machine Learning kernels, some of which
 can be easily and efficiently scaled out to clusters of workstations.
-Internally it uses Intel(R) DAAL (Intel(R) Data Analytics Acceleration
-Library) to get best performance.
+Internally it uses Intel® DAAL (Intel® Data Analytics Acceleration
+Library) to deliver the  best performance.
 
-It's Easy
+Designed for Data Scientists and Framework Designers
 ---------
-A daal4py machine learning algorithm gets constructed with a rich set of
-parameters. Let's assume we want to find the initial set of centroids for kmeans.
-We create an algorithm and configure it for 10 clusters using the 'PlusPlus' method::
+daal4py was created to give data scientists the easist way to utilize Intel® DAAL's
+powerful machine learning building blocks directly in a high-productivity manner.
+A simplified API gives high-level abstractions to the user with minimal boilerplate,
+allowing for quick to write and easy to maintain code when utilizing Jupyter Notebooks.
+For scaling capabilities, daal4py also provides the ability to do distributed machine
+learning, giving a quick way to scale out without having to manually write the
+distributed wiring.
+
+For framework designers, daal4py's has been fashioned to be built under other
+frameworks from both an API and feature perspective.  The machine learning models split
+the training and inference classes, allowing the model to be exported and serialized
+if desired.  This design also gives the flexibility to work directly with the model and
+associated primitives, allowing one to customize the behavior of the model itself.
+The daal4py package can be built with customized algorithm loadouts, allowing for a
+smaller footprint of dependencies when neessary.
+
+Daal4py's Design
+---------
+The design of daal4py utilizes several different technologies to deliver Intel®
+DAAL performance in a flexible design to Data Scientists and Framework designers.
+The package uses Jinja templates to generate Cython-wrapped DAAL C++ headers, with
+Cython as a bridge between the generated DAAL code and the Python layer.
+This design allows for quicker development cycles and acts as a reference design
+to those looking to tailor their build of daal4py.  Cython also allows for better
+Python behavior, both for compatability to different frameworks and for
+pickling and serialization.
+
+API Design and usage
+---------
```

Comment:
All underlines should have same size as headings, getting tons of warnings otherwise.